boj/silver/바이러스/minjong.js

```diff
@@ -0,0 +1,35 @@
+// https://www.acmicpc.net/problem/2606
+
+const [[c], [n], ...lines] = require('fs').readFileSync('../../input.txt', 'utf-8').trim().split('\n').map((x) => x.split(' ').map(Number));
+
+function solution(c, n, lines) {
+  let count = 0;
+  const graph = Array.from({ length: c + 1 }, () => []);
+  const visited = new Array(c + 1).fill(false);
+
+  for (const [src, dst] of lines) {
+    graph[src].push(dst);
+    graph[dst].push(src);
+  }
+
+  const queue = [1];
+
+  while (queue.length !== 0) {
+    const src = queue.shift();
+
+    if (!visited[src]) {
+      visited[src] = true;
+      count += 1;
+
+      for (const dst of graph[src]) {
+        if (!visited[dst]) {
+          queue.push(dst);
+        }
+      }
+    }
+  }
+
+  return count - 1;
+}
```

Comment:
역시 조건이 분리되어 있어서 `indent`가 깊어지는 경향이 있네요!

아래처럼 조건을 하나로 합치면 훨씬 간결할 것 같습니다! 물론, 초기 시작 값(1번 pc)에 대한 `visited`는, 처음부터 업데이트를 해준 상태로 시작해야 합니다.

```suggestion
const queue = [1];
visited[1] = true;

while (queue.length !== 0) {
  const src = queue.shift();

  for (const dst of graph[src]) {
    if (!visited[dst]) {
      queue.push(dst);
      visited[dst] = true;
      count++;
    }
  }
}

return count;
```